sources/api/apiserver/src/server/controller.rs

```diff
@@ -366,7 +366,8 @@ pub(crate) fn set_settings<D: DataStore>(
     transaction: &str,
 ) -> Result<()> {
     trace!("Serializing Settings to write to data store");
-    let pairs = to_pairs_with_prefix("settings", settings)
+    let settings_json = serde_json::to_value(settings).expect("struct to value can't fail");
```

Comment:
Why is this infallible?

Reply:
my rationale on this was that because Settings are serialized through JSON, in order for anything to be assigned to a Settings type it must be serializable to JSON. I guess it's probably safer to handle the error gracefully if it does occur though, so I'll add a snafu error here.